video/out/vo_dmabuf_wayland.c

```diff
@@ -700,16 +701,26 @@ static int reconfig(struct vo *vo, struct mp_image *img)
     if (!vo_wayland_reconfig(vo))
         return VO_ERROR;
 
+    struct gl_video_opts *gl_opts = mp_get_config_group(NULL, vo->global, &gl_video_conf);
+
     mp_mutex_lock(&vo->params_mutex);
     p->target_params = img->params;
+
     // Restore fallback layer parameters if available.
     mp_image_params_restore_dovi_mapping(&p->target_params);
     // Strip metadata that is not understood anyway.
     struct pl_hdr_metadata *hdr = &p->target_params.color.hdr;
     hdr->scene_max[0] = hdr->scene_max[1] = hdr->scene_max[2] = 0;
     hdr->scene_avg = hdr->max_pq_y = hdr->avg_pq_y = 0;
+
+    if (gl_opts->target_prim)
+        p->target_params.color.primaries = gl_opts->target_prim;
+    if (gl_opts->target_trc)
+        p->target_params.color.transfer = gl_opts->target_trc;
```

Comment:
Why not use `-vf=format` to adjust the video/output? `--target-*` options are meant to be used to convert video to given format.

Reply:
Well the idea here is that the compositor would be the one doing the converting.

Reply:
but with this protocol we say what format is the video, not really what is outputted from compositor. Also `vf=format` by default doesn't do any conversions and just overrides video tagged colorspace, so it is perfect for the job here.

Reply:
Yeah you're right. Will drop this.